src/DependencyGraph/ResolutionRequest.js

```diff
@@ -15,6 +15,13 @@ const realPath = require('path');
 const isAbsolutePath = require('absolute-path');
 const getAssetDataFromName = require('../lib/getAssetDataFromName');
 const Promise = require('promise');
+const throat = require('throat')(Promise);
+
+const MAX_CONCURRENT_FILE_READS = 32;
+const getDependencies = throat(
+  MAX_CONCURRENT_FILE_READS,
+  (module, transformOptions) => module.getDependencies(transformOptions)
```

Comment:
Does this end up being recursive?  If `module.getDependencies` ends up calling the throttled `getDependencies`, you could end up in deadlock, which would ultimately fail silently as it's undetectable.  Providing `getDependencies` is not recursive, this looks fine :+1: